### PR TITLE
docs: add backend guide

### PR DESCRIPTION
**Pull request by andreasgangso**
Just to detail some things that can be tricky to wrap head around when first joining the project

### CODE REVIEW

File: docs/backend-guide.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+### Special collections
+
+Collections also back personalized collections like "Continue watching", "shorts" (algorithm-based random shorts), and "My list".
+
+These work exactly like other collections, but has an additional filtering step. Example for "my list":
+
+- Get collection ids like normal, based on the configuration for the collection. This is cached. Let's say it returns IDs: 1,2,3,4,5
+- Get all ids in this specific user's "my_list" (favorites): IDs are 1,3,8,10.
+- You end up with IDs 1,3
```
Comment (KillerX) [inline docs/backend-guide.md:55]: This seems somewhat unclear to/confusing to me. What example are you trying to make?

**Review by KillerX — APPROVED** [review]
It's very high level, but I do see the value if you have 0 context otherwise.

File: docs/backend-guide.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+2. Go to the other end:
+
+- Define your graphql schema for how you want to expose the feature
+
+3. Fill in between:
+
+- Codegen the graphql resolvers
+- Write sql queries to get the data you need.
+- Write [dataloaders]() to get the data
```
Comment (sifferhans) [inline docs/backend-guide.md:18]: Should there be a link here?

File: docs/backend-guide.md (hunk 1 above)
Comment (andreasgangso) [reply docs/backend-guide.md]: yea.. shortened and moved as last point in section above it

**Review by sifferhans — APPROVED** [review]

**Pull request merged**